src/Umbraco.Web/PublishedCache/XmlPublishedCache/XmlPublishedContent.cs

```diff
@@ -432,7 +433,20 @@ private void InitializeNode()
 		    }
 
             //lookup the content type and create the properties collection
-            contentType = getPublishedContentType(PublishedItemType.Content, docTypeAlias);
+            try
+            {
+                contentType = getPublishedContentType(PublishedItemType.Content, docTypeAlias);
+
+            }
+            catch (Exception e)

```

Comment:
Actually I'm going to change this so we don't need to be checking for TargetSite (which can be null in some cases) this is precisely why you shouldn't be throwing generic exception types and choose a specific type so you can catch it and deal with it correctly.